### PR TITLE
docs: fix PHPDoc types in `Route` class.

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
See #6310 

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Router/RouteCollectionInterface.php [hunk 1]
```diff
@@ -30,10 +30,8 @@ interface RouteCollectionInterface
      *
      * @param array|Closure|string $to
      * @param array                $options
-     *
-     * @return mixed
      */
-    public function add(string $from, $to, ?array $options = null);
+    public function add(string $from, $to, ?array $options = null): self;
```
Comment (kenjis) [inline system/Router/RouteCollectionInterface.php:34]: This is a breaking change.
https://3v4l.org/n9AYI

**Review by MGatner — CHANGES_REQUESTED** [review]
Please adjust this to docblock changes only

**Comment by ping-yee** [comment]
OK, I'll change it later.

**Comment by ping-yee** [comment]
I have been fixed.

File: system/Router/RouteCollectionInterface.php (hunk 1 above)
Comment (ping-yee) [reply system/Router/RouteCollectionInterface.php]: I have been removed.

**Review by MGatner — APPROVED** [review]
Thank you!

**Pull request merged**